File: PB - More Exercises Python/2.0 Conditional Statements - More Exercises/03. Harvest.py
from math import ceil, floor, trunc

vineyard, kvM, necessary_wine, workers = [float(input()) for _ in range(4)]

all_wine = (vineyard - (vineyard * 0.60)) * kvM / 2.5
bonus_wine = all_wine - necessary_wine
wine_for_workers = bonus_wine / workers

if necessary_wine > all_wine:
    print(f'It will be a tough winter! More {abs(trunc(bonus_wine))} liters wine needed.')
else:
    print(f'Good harvest this year! Total wine: {floor(all_wine)} liters.\n\
{ceil(bonus_wine)} liters left -> {ceil(wine_for_workers)} liters per person.')


'''
650
2
175
3
----------
1020
1.5
425
4

'''

File: PB - More Exercises GO/2.0 Conditional Statements - More Exercises/03. Harvest.go
package main

import "fmt"
import "math"

func main() {
	var vineyard, kvM, necessaryWine, workers float64
	fmt.Scan(&vineyard, &kvM, &necessaryWine, &workers)
	vineyard *= 0.40
	allWine := vineyard * kvM / 2.5
	bonusWine := allWine - necessaryWine
	wineForWorkers := bonusWine / workers

	if necessaryWine > allWine {
		fmt.Printf("It will be a tough winter! More %v liters wine needed.", math.Abs(math.Trunc(bonusWine)))
	} else {
		fmt.Printf("Good harvest this year! Total wine: %v liters.\n%v liters left -> %v liters per person.",
			math.Floor(allWine), math.Ceil(bonusWine), math.Ceil(wineForWorkers))
	}

}

/*
name   :03. Harvest
input  :650 2 175 3
output :
Good harvest this year! Total wine: 208 liters.
33 liters left -> 11 liters per person.
*/
